### PR TITLE
Add support for Join type.

### PR DESCRIPTION
**Pull request by rishi-aga**
Add support for Join type.
-- Rename existing `type` property to `kind`
-- Add `type` property

## How Has This Been Tested?
Existing tests pass.

## License
I confirm that this contribution is made under an Apache 2.0 license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/annotation/Join.java [hunk 1]
```diff
@@ -21,4 +21,10 @@
      * @return join constraint like <code>{{col1}} = {{joinField.col2}}</code>
      */
     String value();
+
+    /**
+     * Join type.
+     * @return join type like {@code left, right, inner, outer or cross}
+     */
+    String type() default "left";
```
Comment (aklish) [inline elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/annotation/Join.java:29]: Let's use the enum here.

File: elide-model-config/src/main/java/com/yahoo/elide/modelconfig/jsonformats/ElideJoinTypeFormatAttr.java [hunk 2]
```diff
@@ -15,15 +15,16 @@
 /**
  * Format specifier for {@code elideJoinType} format attribute.
  * <p>
- * This specifier will check if a string instance is one of {@code ToOne, ToMany}.
+ * This specifier will check if a string instance is one of {@code left, right, inner, outer, cross}.
  * </p>
  */
 public class ElideJoinTypeFormatAttr extends AbstractFormatAttribute {
-    private static final String JOIN_TYPE_REGEX = "^(?i)(ToOne|ToMany)$";
+    private static final String JOIN_TYPE_REGEX = "^(?i)(left|right|inner|outer|cross)$";
```
Comment (aklish) [inline elide-model-config/src/main/java/com/yahoo/elide/modelconfig/jsonformats/ElideJoinTypeFormatAttr.java:22]: I think we should have left, inner, full, and cross as the join types.

File: elide-model-config/src/main/java/com/yahoo/elide/modelconfig/model/Join.java [hunk 3]
```diff
@@ -39,16 +40,39 @@
...
+            return this.value;
+        }
+    }
+
+    public enum Type {
+
+        LEFT("left"),
+        RIGHT("right"),
+        INNER("inner"),
+        OUTER("outer"),
```
Comment (aklish) [inline elide-model-config/src/main/java/com/yahoo/elide/modelconfig/model/Join.java:71]: See earlier comment about supported join types.

File: elide-model-config/src/main/resources/elideTableSchema.json [hunk 4]
```diff
@@ -33,7 +39,7 @@
             "required": [
                 "name",
                 "to",
-                "type",
+                "kind",
```
Comment (aklish) [inline elide-model-config/src/main/resources/elideTableSchema.json:42]: Let's make this optional and default to toOne.

File: elide-model-config/src/test/resources/validator/valid/models/tables/player_stats.hjson [hunk 5]
```diff
@@ -23,13 +23,14 @@
...
-             type: toOnE
+             kind: toOnE
+             type: outeR
              definition: '{{ playerCountry.id}} = {{country_id}}'
           },
           {
              name: playerTeam
              to: Team
-             type: Tomany
+             kind: Tomany
```
Comment (aklish) [inline elide-model-config/src/test/resources/validator/valid/models/tables/player_stats.hjson:33]: What about SQL generation tests for these different joins.  Are there any dialect differences?

**Review by aklish — CHANGES_REQUESTED** [review]

File: elide-model-config/src/test/resources/validator/valid/models/tables/player_stats.hjson (hunk 5 above)
Comment (rishi-aga) [reply elide-model-config/src/test/resources/validator/valid/models/tables/player_stats.hjson:32]: placeholder changes only  in this PR, will add support for these join types in another PR>

File: elide-model-config/src/main/resources/elideTableSchema.json (hunk 4 above)
Comment (rishi-aga) [reply elide-model-config/src/main/resources/elideTableSchema.json]: done.

File: elide-model-config/src/main/java/com/yahoo/elide/modelconfig/model/Join.java (hunk 3 above)
Comment (rishi-aga) [reply elide-model-config/src/main/java/com/yahoo/elide/modelconfig/model/Join.java]: done.

File: elide-model-config/src/main/java/com/yahoo/elide/modelconfig/jsonformats/ElideJoinTypeFormatAttr.java (hunk 2 above)
Comment (rishi-aga) [reply elide-model-config/src/main/java/com/yahoo/elide/modelconfig/jsonformats/ElideJoinTypeFormatAttr.java]: done.

File: elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/annotation/Join.java (hunk 1 above)
Comment (rishi-aga) [reply elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/annotation/Join.java]: done.

**Review by aklish — APPROVED** [review]

**Pull request merged**